app/services/remember_device_cookie.rb

```diff
@@ -0,0 +1,30 @@
+class RememberDeviceCookie
+  attr_reader :user_id, :created_at
+
+  def initialize(user_id:, created_at:)
+    @user_id = user_id
+    @created_at = created_at
+  end
+
+  def self.from_json(json)
+    parsed_json = JSON.parse(json)
```

Comment:
Here's where we'd reject this if the parsed JSON didn't have our sentinel indicating this is from a proper remember device cookie.